proxy/http/HttpTransactCache.cc

```diff
@@ -1299,7 +1299,7 @@ HttpTransactCache::match_response_to_request_conditionals(HTTPHdr *request, HTTP
       ink_time_t lm_value = response->get_last_modified();
 
       // we won't return NOT_MODIFIED if Last-modified is too recent
-      if ((lm_value == 0) || (request->get_if_modified_since() < lm_value)) {
```

Comment:
Another acceptable change would be to only allow this optimization if the response is cacheable.  In your use case the response from the origin is not cacheable.